src/steamship/agents/examples/fact_learner.py

```diff
@@ -0,0 +1,75 @@
+import uuid
+from typing import List
+
+from steamship import Block
+from steamship.agents.llms.openai import OpenAI
+from steamship.agents.react import ReACTAgent
+from steamship.agents.schema import AgentContext
+from steamship.agents.schema.context import Metadata
+from steamship.agents.service.agent_service import AgentService
+from steamship.agents.tools.question_answering import VectorSearchQATool
+from steamship.agents.tools.question_answering.vector_search_learner_tool import (
+    VectorSearchLearnerTool,
+)
+from steamship.agents.utils import with_llm
+from steamship.invocable import post
+from steamship.utils.repl import AgentREPL
+
+
+class FactLearner(AgentService):
+    """FactLearner is an example AgentService contains an Agent which:
+
+    1) Learns facts to a vector store
+    2) Can answer questions based on those facts"""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._agent = ReACTAgent(
+            tools=[
+                VectorSearchLearnerTool(),
+                VectorSearchQATool(),
+            ],
+            llm=OpenAI(self.client),
+        )
+
+    @post("prompt")
+    def prompt(self, prompt: str) -> str:
+        """Run an agent with the provided text as the input."""
+
+        # AgentContexts serve to allow the AgentService to run agents
+        # with appropriate information about the desired tasking.
+        # Here, we create a new context on each prompt, and append the
+        # prompt to the message history stored in the context.
+        context_id = uuid.uuid4()
+        context = AgentContext.get_or_create(self.client, {"id": f"{context_id}"})
+        context.chat_history.append_user_message(prompt)
+
+        # Add the LLM
+        context = with_llm(context=context, llm=OpenAI(client=self.client))
```

Comment:
question: would this be a good place to play with different temp settings?

Reply:
Unsure -- what would the nature of the playing be in this case? To show how different temps alter the way in which it answers the questions?